lib/px/utils/pxcommonutils.lua

```diff
@@ -153,14 +153,68 @@ function _M.extract_cookie_names(cookies)
     return t
 end
 
-function _M.call_px_server(httpc, host, port, proxy)
-    if proxy ~= nil then
-        return httpc:connect_proxy(proxy, "https", host, port, "")
+function _M.call_px_server(httpc, host, port, px_config, pool_key)
+    if px_config.proxy_url ~= nil then
+        local scheme = px_config.ssl_enabled and "https" or "http"
+        local proxy_authorization = px_config.proxy_authorization == nil and "" or px_config.proxy_authorization
```

Comment:
why do we need to do that here? I thought when we set the px configuration we set the default or you what the customer configured - we can set empty string there.

Reply:
yea the default is nil...